src/main/java/com/google/googleidentity/security/UserSession.java

```diff
@@ -0,0 +1,43 @@
+package com.google.googleidentity.security;
+
+import com.google.googleidentity.user.UserDetails;
+import com.google.inject.servlet.SessionScoped;
+
+import java.io.Serializable;
+
+@SessionScoped
+public class UserSession implements Serializable {
+
+    private static final long serialVersionUID = 3L;
+
+    private UserDetails user = null;
```

Comment:
nit: perfer java.util.Optional to null.

```suggestion
    private Optional<UserDetails> user = Optional.empty();
```

Reply:
It seems Optional type can not be used as a class member here for an @SessionScoped class since it is not Serializable.